src/lib.rs

```diff
@@ -15,6 +16,34 @@ struct Bloom {
     hash_func: Option<PyObject>,
 }
 
+#[derive(Debug)]
+enum SaveLoadType {
+    Filepath(String),
+    FileLike(PyFileLikeObject),
+    Bytes(Vec<u8>),
+    None,
+}
```

Comment:
Would recommend separate types for Save/Load, since e.g. one accepts Bytes, the other doesn't, and we also different requirements for the file-like object (readable vs writable). We can also use pyo3's derive for FromPyObject:

```rust
fn extract_readable_file(obj: &PyAny) -> PyResult<PyFileLikeObject> {
    PyFileLikeObject::with_requirements(obj.into(), true, false, false)
}

fn extract_writable_file(obj: &PyAny) -> PyResult<PyFileLikeObject> {
    PyFileLikeObject::with_requirements(obj.into(), false, true, false)
}

#[derive(Debug, FromPyObject)]
enum SaveDest {
    Filepath(PathBuf),
    FileLike(#[pyo3(from_py_with = "extract_writable_file")] PyFileLikeObject),
}

#[derive(Debug, FromPyObject)]
enum LoadSrc<'a> {
    Filepath(PathBuf),
    FileLike(#[pyo3(from_py_with = "extract_readable_file")] PyFileLikeObject),
    Bytes(&'a [u8]),
}
```